### PR TITLE
Upgrading file copy methods in weather-sp to use gsutil.

### PR DESCRIPTION
**Pull request by alxmrs**

### CODE REVIEW

File: weather_sp/splitter_pipeline/file_splitters.py [hunk 1]
```diff
@@ -225,10 +228,9 @@ def _write_dataset(self, dataset: xr.Dataset, split_dims: t.List[str]) -> None:
         # returning bytes. Further, the scipy engine does not support NETCDF4 (which is HDF5 compliant).
         # Storing data in HDF5 is advantageous since it allows opening NetCDF files with buffered readers.
         with tempfile.NamedTemporaryFile() as tmp:
-            with FileSystems().create(self._get_output_for_dataset(dataset, split_dims)) as dest_file:
-                dataset.to_netcdf(path=tmp.name, engine='netcdf4', format='NETCDF4')
-                tmp.seek(0)
-                shutil.copyfileobj(tmp, dest_file, WRITE_CHUNK_SIZE)
+            dataset.to_netcdf(path=tmp.name, engine='netcdf4', format='NETCDF4')
+            tmp.seek(0)
```
Comment (fredzyda) [inline weather_sp/splitter_pipeline/file_splitters.py:232]: Why seek 0 here? Does this file object not have a flush method or somilar?

**Review by fredzyda — CHANGES_REQUESTED** [review]
Why use subprocess when there is an existing python library for this?

File: weather_sp/splitter_pipeline/file_splitters.py (hunk 1 above)
Comment (alxmrs) [reply weather_sp/splitter_pipeline/file_splitters.py:232]: Good catch! This is left over from the previous implementation.

File: weather_sp/splitter_pipeline/file_splitters.py [hunk 2]
```diff
@@ -26,16 +26,23 @@
...
 from .file_name_utils import OutFileInfo
 
 logger = logging.getLogger(__name__)
 
 
+# TODO(#245): Group with common utilities (duplicated)
+def copy(src: str, dst: str) -> None:
+    """Copy data via `gsutil cp`."""
+    try:
+        subprocess.run(['gsutil', 'cp', src, dst], check=True, capture_output=True)
```
[the remark this answers is not in the document]
Comment (alxmrs) [reply weather_sp/splitter_pipeline/file_splitters.py:40]: Discussed offline. TL;DR: We've benchmarked each implementation, and found that this was the fastest approach, given our current dependencies: 

https://github.com/google/weather-tools/issues/256#issuecomment-1338482275

[the remark this answers is not in the document]
Comment (fredzyda) [reply weather_sp/splitter_pipeline/file_splitters.py:40]: OK, annoying, but I agree it is like double the speed.

**Review by fredzyda — APPROVED** [review]

**Pull request merged**